mirror.go

```diff
@@ -175,7 +182,12 @@ func (m *mirror) pullImage(tag string) error {
 		authConfig.Password = os.Getenv("DOCKERHUB_PASSWORD")
 	}
 
-	return m.dockerClient.PullImage(pullOptions, authConfig)
+	if m.repo.PrivateRegistry != "" {
+		pullOptions.Repository = m.repo.PrivateRegistry + m.repo.Name
+		return (*m.dockerClient).PullImage(pullOptions, authConfig)
+	}
+
+	return (*m.dockerClient).PullImage(pullOptions, authConfig)
```

Comment:
Why the parenthesis?

Reply:
Updated this, removed a pointer in favor of passing the docker client interface in directly to the monitor struct, this allows us to not have to reference things like this. The parenthesis dereferenced the function to allow it to be used after pointing to the dockerclient.